pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1191,6 +1191,12 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 			newDisk.Driver.IOThread = &ioThreadId
 		}
 
+		err = SetDriverCacheMode(&newDisk)
```

Comment:
SetDriverCacheMode checks the disk location, if it supports directio. It is a function with a side-effect. Could you instead of just calling this function here, split the function apart? It would be great if we determine outside of the converter which locations support this flag, and pass that information via the `ConverterContext`. No side-effects allows us easy testing  of the converter.

It should then be easy to adjust unit tests for it.

Reply:
So expand ConverterContext and move the calls for SetDriverCacheMode into SyncVMI and PrepareMigrationTarget?

Reply:
I haven't tested these changes yet but I want to make sure I'm going in the right direction.

Reply:
It looks like I still have to set the cache inside of prestarthook too. I'm testing against my branch of v.0.34.0 as master is having issues with the build environment.

Reply:
I'm starting to think I should just call the original versions of the Set after the Convert_v1_VirtualMachine_To_api_Domain calls in the relevant functions in manager.go.

Reply:
I guess it makes sense for a focused fix.

Reply:
domain.Spec.Devices.Disks isn't populated until Convert_v1_VirtualMachine_To_api_Domain is called. This seemed better than trying to unwind the vmi volume logic to run this before the conversion and set new elements of ConvertContext.

Reply:
I am fine with your approach. I just don't like that we can get out of sync again to a later point in time. It is easy to think that just running the converter fully converts the VMI to the domain (which it actually should do in an ideal  world).